Clicker/Models/ListDiffableModels/Poll.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Poll.swift
+//  Clicker
+//
+//  Created by Kevin Chan on 2/3/18.
+//  Copyright © 2018 CornellAppDev. All rights reserved.
+//
+
+import UIKit
+import IGListKit
+
+enum PollState {
+    case live
+    case ended
+    case shared
+}
+
+class Poll {
+    
+    var id: Int?
+    var text: String
+    var questionType: QuestionType
+    var options: [String]
+    var results: [String:Any]
+    // results format:
+    // MULTIPLE_CHOICE: {'A': {'text': 'Blue', 'count': 3}, ...}
+    // FREE_RESPONSE: {'Blue': {'text': 'Blue', 'count': 3}, ...}
+    var state: PollState!
+    
+    // MARK: - Constants
+    let defaultIdentifier = UUID().uuidString
+    let idKey = "key"
+    let textKey = "text"
+    let countKey = "count"
+    let optionsKey = "options"
+    let typeKey = "type"
+    
+    // MARK: SORTED BY DATE POLL INITIALIZER
+    init(id: Int, text: String, results: [String:Any], type: QuestionType, state: PollState) {
+        self.id = id
+        self.text = text
+        self.results = results
+        self.options = results.map { (key, _) in key }
+        self.questionType = type
+        self.state = state
+    }
+    
+    // MARK: SEND START POLL INITIALIZER
+    init(text: String, options: [String], type: QuestionType, state: PollState) {
+        self.text = text
+        self.options = options
+        self.results = [:]
+        self.questionType = type
+        for (index, option) in options.enumerated() {
```

Comment:
thoughts on this vs `options.enumerated().forEach {...`

Reply:
I like the `forEach` :)